packages/components/src/border-box-control/test/index.tsx

```diff
@@ -189,7 +188,14 @@ describe( 'BorderBoxControl', () => {
 
 			const colorButton = screen.getByLabelText( colorPickerRegex );
 			await user.click( colorButton );
-			await act( () => Promise.resolve() );
+
+			await waitFor( () =>
+				expect(
+					screen.getByRole( 'button', {
+						name: 'Custom color picker.',
+					} )
+				).toBeVisible()
+			);
```

Comment:
@tyxla  this change is actually triggering the well-know "An update to %s inside a test was not wrapped in act(...)." warning.

Any ideas on how to fix it? I tried following your approach from other PRs but couldn't make it work.

Thank you!

Reply:
I realised that I had forgotten to stop using fake timers — did that in https://github.com/WordPress/gutenberg/pull/47755/commits/ca41ab04bde65239566e0836fcb031b83b9d0dc8. It should hopefully fix the issue 🤞

Reply:
It still looks like there may be timeout issues — I'd appreciate your help in understanding how to fix those errors. Thank you!

Reply:
Sure thing! I've filed a commit and will follow up with a comment to explain.

Reply:
Real timers fixed it locally, but your commit https://github.com/WordPress/gutenberg/pull/47755/commits/bde3e094416d2ddb0683eb5b16c923fde117090d fixed on CI as well. Thank you 🙏